service-front/module/Application/test/Controller/CPFlowControllerTest.php

```diff
@@ -269,4 +274,23 @@ public function testPostOfficeCountriesPage(): void
         $this->assertControllerClass('CpFlowController');
         $this->assertMatchedRouteName('root/cp_choose_country');
     }
+
+    public function testPostOfficeCountriesIdPage(): void
```

Comment:
The only thing this test really checks for is that `getDetailsData` is called. It should also cover the logic in the controller code: what's displayed on the page, what happens on submission, validation, saving the data to the API (or those things should be mocked out and tested in smaller units)

Reply:
All the controller function tests do this: they simply check that the route loads the page without errors and nothing more. I understood that we were going to use cypress for testing content and behaviour?

Reply:
We try to only use Cypress for happy path smoke tests, because it's so much slower and harder to run/debug than PHPUnit tests (per the [Test Pyramid](https://martinfowler.com/bliki/TestPyramid.html)).

Appreciate we've not been treating tests as a high priority, but as the application gets more complicated we're running into more regressions so we really need to build up our test coverage going forward.